### PR TITLE
Replace Quill with Slick

### PR DESCRIPTION
**Pull request by marialivia16**

### CODE REVIEW

**Comment by marialivia16** [comment]
[Switch from Quill to Slick](https://trello.com/c/Vzrn8kSe/71-switch-from-quill-to-slick)

File: app/AppComponents.scala [hunk 1]
```diff
@@ -19,18 +22,19 @@ class AppComponents(context: Context)
...
   //Context is created here so we can add a stop hook to kill the db connection when the app terminates
-  lazy val dbContext = new PostgresJdbcContext[SnakeCase](config.dbConfig)
-  lazy val db = new MetricsDB(dbContext)
+  private lazy val dbConfig: DatabaseConfig[PostgresProfile] = DatabaseConfig.forConfig("slick.dbs.default", configuration.underlying)
+  private lazy val db: Database = dbConfig.db
+  lazy val metricsDb = new MetricsDB(db)
 
   lazy val kinesisStreamConsumer = new ProductionMetricsStreamReader(config.publishingMetricsKinesisStream, config.stage, config)
-  kinesisStreamConsumer.start
+  kinesisStreamConsumer.start()
```
Comment (susiecoleman) [inline app/AppComponents.scala]: Do you need the `()`

File: app/models/db/Metrics.scala [hunk 2]
```diff
@@ -9,16 +9,22 @@ import org.joda.time.DateTime
 
 case class Metric(
     id: String,
-    startingSystem: String,
+    startingSystem: OriginatingSystem,
```
Comment (susiecoleman) [inline app/models/db/Metrics.scala]: Can`startingSystem` be renamed to `originatingSystem`

File: app/models/db/MetricsFilters.scala [hunk 3]
```diff
@@ -1,59 +1,68 @@
...
-                           desk: Option[String],
-                           startingSystem: Option[String] = None,
-                           filtersList: List[String]
+  dateRange: Option[DateRange] = None,
+  desk: Option[String] = None,
+  originatingSystem: Option[OriginatingSystem] = None
 )
 
 object MetricsFilters {
+  private val TrueOptCol : Rep[Option[Boolean]] = LiteralColumn(Some(true))
```
Comment (susiecoleman) [inline app/models/db/MetricsFilters.scala]: I don't understand what this does... Is it a slick thing?

File: app/models/db/MetricsFilters.scala [hunk 4]
```diff
@@ -1,59 +1,68 @@
...
 
     dateRange match {
       case None =>
-        val start = getOptionFromQS("startDate", qs).flatMap(parseDate)
-        val end = getOptionFromQS("endDate", qs).flatMap(parseDate)
-
-        if(start.isEmpty && end.isEmpty) None
-        else Some(DateRange(from = start, to = end))
+        Some(DateRange(
+          from = getOptionFromQS("startDate", qs).flatMap(parseDate).getOrElse(DateTime.now().minusYears(10)),
```
Comment (susiecoleman) [inline app/models/db/MetricsFilters.scala]: Why minus 10 years?

File: app/models/db/Schema.scala [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  val metricsTable: TableQuery[DBMetric] = TableQuery[DBMetric]
+  val inCopyMetricsTable: TableQuery[DBInCopyMetric] = TableQuery[DBInCopyMetric]
+  val composerMetricsTable: TableQuery[DBComposerMetric] = TableQuery[DBComposerMetric]
+  val forksTable: TableQuery[DBFork] = TableQuery[DBFork]
+
+  class DBMetric(tag: Tag) extends Table[Metric](tag, "metrics") {
+    private implicit val originatingSystemEncDec = MappedColumnType.base[OriginatingSystem, String](os => os.entryName, name => OriginatingSystem.withName(name))
+
+    def id                  = column[String]("id", O.PrimaryKey)
+    def startingSystem      = column[String]("starting_system")
```
Comment (marialivia16) [inline app/models/db/Schema.scala]: Yes, sounds reasonable.

File: app/AppComponents.scala (hunk 1 above)
Comment (marialivia16) [inline app/AppComponents.scala]: IntelliJ did it.

File: app/models/db/Metrics.scala (hunk 2 above)
Comment (marialivia16) [inline app/models/db/Metrics.scala]: I'll rename everything when I apply the new evolution.

File: app/models/db/MetricsFilters.scala (hunk 3 above)
Comment (marialivia16) [inline app/models/db/MetricsFilters.scala]: Yes, it's creating a boolean that the db understands.

File: app/AppComponents.scala (hunk 1 above)
Comment (susiecoleman) [inline app/AppComponents.scala]: Sneaky Intellij

File: app/models/db/Metrics.scala (hunk 2 above)
Comment (susiecoleman) [inline app/models/db/Metrics.scala]: 👍

File: app/models/db/MetricsFilters.scala (hunk 4 above)
Comment (marialivia16) [inline app/models/db/MetricsFilters.scala]: Because I don't know how to say "the beginning of times".

Comment (susiecoleman) [inline app/models/db/MetricsFilters.scala]: Ok cool. Maybe pull `DateTime.now().minusYears(10)` out into a variable and call it something like `baseMetricsDate`? Or a better name than that

**Comment by susiecoleman** [comment]
I can have a look again if you're going to do the evolutions in this PR but looks good

File: app/models/db/MetricsFilters.scala (hunk 4 above)
Comment (mbarton) [inline app/models/db/MetricsFilters.scala]: I think `new DateTime(0)` will get you the unix epoch start

Comment (susiecoleman) [inline app/models/db/MetricsFilters.scala]: Yep that works 🐫 cheers

**Pull request merged**